defaulting-capi-clusters/README.md

```diff
@@ -0,0 +1,46 @@
+# Defaulting of CAPI clusters with webhooks
+
+This RFC describes a pattern for defaulting of CAPI clusters by utilizing Kubernetes webhooks extensively.
+
+## Problem statement
+
+We want clusters to be created with defaulted values.
+These values should match the clusters in our current product reasonably well.
+Therefore we will need to default some values differently from upstream Cluster-API.
+
+We do not expect to have full control of the method of cluster creation in the future.
+Therefore we have to account for different methods of cluster creation ( e.g. kubectl, kubectl-gs, happa, ...) while keeping defaulting consistent between those methods.
```

Comment:
💯

Reply:
let's add gitops as a method as the saving of dry run manifests in git is a use case we definitely need to support for several customers already. this has a bit of impact on the webhook requirements you mention below